### ISSUE TITLE
Compilation problem does not report correctly when running in Teamcity (& likely other CIs)

### ISSUE THREAD

**Issue by lilianaziolek** [nuxt/nuxt.js#5951]
### Version

[v2.8.1](https://github.com/nuxt.js/releases/tag/v2.8.1)

### Reproduction link

[https://codesandbox.io/s/codesandbox-nuxt-hvv2n](https://codesandbox.io/s/codesandbox-nuxt-hvv2n)

### Steps to reproduce

1. Create a default nuxt project and create a broken component - for example importing non-existing component (or download linked CodeSandbox project)

2. Run to check for expected behaviour
yarn install && yarn build && echo "should not see this"
You will get nuxt fatal error and "should not see this" will not be printed.

3. You can run same project in Teamcity to see build would succeed and no logs printed, but easier is to simulate this by setting env variable, e.g.
export BUILD_NUMBER=123

4. Clean up the directory and run the same command as 2. Rather than seeing build errors and skipped last command, the build will return 0 and will not print that it failed.

### What is expected ?

1. Even in quiet mode, I would expect the build to print compilation errors
2. Build should fail with result code > 0

### What is actually happening?

build quietly fails and returns 0 ("should not see this" is printed)

### Additional comments?

As a current workaround, setting quiet:false in build section in nuxt.config.js enforces the behaviour in Teamcity to be same as usual, i.e. reports issues correctly in logs and with error code.

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c9376">#c9376</a>)</em></sub></div>

**Comment by lilianaziolek** [comment]
This issue still occurs. To reiterate reproduce steps:
1. Download the codesandbox content as ZIP.
2. Simulate running in TeamCity: go to command line, export env variable (use set on windows)
```
export BUILD_NUMBER=123
```
3. Run command to install dependencies, run build and check status:
```
yarn install && yarn build && echo "should not see this"
```
Build status returns 0 and no errors are reported, instead of failing as supposed to.

**Comment by garrensweet** [comment]
This should not be a stale issue. More time was spent on this than should have been necessary to figure out `quiet: false` would actually return the error.

**Comment by garrensweet** [comment]
BUMP. Reproduction is above. This is not a stale issue. The same issue still exists, nothing has changed. There needs to be one (or more) of the following in place:

1. Notify the user that quiet: true is enabled, and that errors will return `0` which DOES NOT FAIL. 
2. set the default to quiet: false and notify the user that errors will cause failures.

Either one is sufficient.

This is still the same issue in edge.

**Comment by lilianaziolek** [comment]
Could someone from Nuxt team comment on this at least?
As far as I can see this is still present, meaning any Nuxt builds in CI environments are potentially unsafe. The fact that build does not report failure means that where there is automated deployment (commit -> CI build & test -> deploy), a broken app can be deployed as CI has no way of seeing that there was a breaking change. In fact this is exactly what happened to us.

There is a workaround, but it's not enabled by default. This issue needs to be fixed (IMO quiet should only impact verbosity of the logging, not build status reporting), or if that's too difficult/too low priority, at least the default could be changed so that CI works with default settings.

**Comment by manniL** [comment]
see #6075

**Comment by lilianaziolek** [comment]
Ahhh, good news :) Thanks, didn't notice the other issue.

**Comment by manniL** [comment]
NP and sorry for the delay :+1: 	

Closing here as it'll be fixed in the upcoming v2.9 (should be already present in `nuxt-edge`)
